File: src/api/ecosystem.ts
import { apiRequest } from "./client";

export type EcosystemStatus =
  | "connected"
  | "read_only"
  | "disabled"
  | "planned"
  | "roadmap"
  | "disconnected"
  | "degraded";

export type EcosystemCheck = {
  id: string;
  status: EcosystemStatus;
  source: string;
  updated_at: string;
  latency_ms?: number | null;
  detail?: string | null;
  error?: string | null;
};

export type HtxEcosystemStatus = {
  api_environment: "production";
  live_trading_status: "disabled";
  account_read_only_status: "planned";
  checks: EcosystemCheck[];
  last_sync: string;
  source: string;
  is_mock: boolean;
};

export type AiComputeCapability = {
  id: string;
  status: EcosystemStatus;
  provider?: string | null;
  model?: string | null;
  detail?: string | null;
};

export type AiComputeStatus = {
  current_provider: string;
  current_model?: string | null;
  current_provider_status: EcosystemStatus;
  credential_status: "configured" | "not_configured";
  connection_status: EcosystemStatus;
  last_tested_at?: string | null;
  planned_provider: "B.AI";
  planned_provider_status: "planned";
  capabilities: AiComputeCapability[];
  updated_at: string;
  is_mock: boolean;
};

export type UtilityTier = {
  id: string;
  status: EcosystemStatus;
  features: Record<string, EcosystemStatus>;
};

export type UtilityModel = {
  tiers: UtilityTier[];
  updated_at: string;
  source: string;
  is_mock: boolean;
};

export type RoadmapItem = {
  id: string;
  status: EcosystemStatus;
  target?: string | null;
  detail?: string | null;
};

export type EcosystemRoadmap = {
  items: RoadmapItem[];
  updated_at: string;
  source: string;
  is_mock: boolean;
};

export function fetchHtxEcosystemStatus(signal?: AbortSignal) {
  return apiRequest<HtxEcosystemStatus>("/api/ecosystem/htx-status", { signal });
}

export function fetchAiComputeStatus(signal?: AbortSignal) {
  return apiRequest<AiComputeStatus>("/api/ecosystem/ai-compute-status", { signal });
}

export function fetchUtilityModel(signal?: AbortSignal) {
  return apiRequest<UtilityModel>("/api/ecosystem/utility-model", { signal });
}

export function fetchEcosystemRoadmap(signal?: AbortSignal) {
  return apiRequest<EcosystemRoadmap>("/api/ecosystem/roadmap", { signal });
}

File: backend/app/schemas/ecosystem.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


EcosystemStatus = Literal["connected", "read_only", "disabled", "planned", "roadmap", "disconnected", "degraded"]


class EcosystemCheck(BaseModel):
    id: str
    status: EcosystemStatus
    source: str
    updated_at: datetime
    latency_ms: int | None = None
    detail: str | None = None
    error: str | None = None


class HtxEcosystemStatus(BaseModel):
    api_environment: Literal["production"]
    live_trading_status: Literal["disabled"]
    account_read_only_status: Literal["planned"]
    checks: list[EcosystemCheck]
    last_sync: datetime
    source: str = "htx_rest"
    is_mock: bool = False


class HtxEcosystemStatusResponse(BaseModel):
    ok: Literal[True] = True
    data: HtxEcosystemStatus


class AiComputeCapability(BaseModel):
    id: str
    status: EcosystemStatus
    provider: str | None = None
    model: str | None = None
    detail: str | None = None


class AiComputeStatus(BaseModel):
    current_provider: str
    current_model: str | None = None
    current_provider_status: EcosystemStatus
    credential_status: Literal["configured", "not_configured"]
    connection_status: EcosystemStatus
    last_tested_at: datetime | None = None
    planned_provider: Literal["B.AI"] = "B.AI"
    planned_provider_status: Literal["planned"] = "planned"
    capabilities: list[AiComputeCapability]
    updated_at: datetime
    is_mock: bool = False


class AiComputeStatusResponse(BaseModel):
    ok: Literal[True] = True
    data: AiComputeStatus


class UtilityTier(BaseModel):
    id: str
    status: EcosystemStatus
    features: dict[str, EcosystemStatus]


class UtilityModel(BaseModel):
    tiers: list[UtilityTier]
    updated_at: datetime
    source: str = "roadmap"
    is_mock: bool = False


class UtilityModelResponse(BaseModel):
    ok: Literal[True] = True
    data: UtilityModel


class RoadmapItem(BaseModel):
    id: str
    status: EcosystemStatus
    target: str | None = None
    detail: str | None = None


class EcosystemRoadmap(BaseModel):
    items: list[RoadmapItem]
    updated_at: datetime
    source: str = "roadmap"
    is_mock: bool = False


class EcosystemRoadmapResponse(BaseModel):
    ok: Literal[True] = True
    data: EcosystemRoadmap
